penguin/llm/adapters/anthropic.py

```
@@ -1086,7 +1215,36 @@ def format_messages(self, messages: List[Dict[str, Any]]) -> List[Dict[str, Any]
+                        if not isinstance(tool_call, dict):
+                            continue
+                        function_payload = tool_call.get("function")
+                        if not isinstance(function_payload, dict):
+                            continue
+                        name = function_payload.get("name")
+                        if not name:
+                            continue
+                        raw_arguments = function_payload.get("arguments") or "{}"
+                        try:
+                            tool_input = json.loads(raw_arguments)
+                        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)